### PR TITLE
fix(provision): redo parameters naming

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/SL9O0bqD/4111-rename-guaranteed-nad-nonguaranteed

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Successfully passed test - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity/543

File: sdcm/provision/common/provisioner.py [hunk 1]
```diff
@@ -8,26 +8,13 @@
...
-    Price for such instances are usually higher
-    """
-
-
-class ProvisionNotGuaranteedParams(ProvisionParamsBase):  # pylint: disable=too-few-public-methods
-    """
-    Provision request parameters for instances that COULD BE REVOKED by cloud provider
-    Price for such instances are usually lower
-    """
+    revocable: bool  # Signals on weather instance could be revoked by cloud provider
```
Comment (bentsi) [inline sdcm/provision/common/provisioner.py:15]: - is it spot?
- weather -> whether

Comment (dkropachev) [reply sdcm/provision/common/provisioner.py:15]: In aws terminology revocable+price = spot

Comment (bentsi) [reply sdcm/provision/common/provisioner.py]: per our conversation rename revocable -> spot

Comment (dkropachev) [reply sdcm/provision/common/provisioner.py]: done

**Comment by dkropachev** [comment]
Tested - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-byo-longevity/544/

**Review by bentsi — APPROVED** [review]

**Pull request merged**